src/main/java/net/revelc/code/formatter/xml/lib/FormattingPreferences.java

```diff
@@ -16,11 +16,16 @@
 package net.revelc.code.formatter.xml.lib;
 
 public class FormattingPreferences {
+    public static final String IGNORE = "IGNORE";
+    public static final String WARN = "WARN";
+    public static final String FAIL = "FAIL";
```

Comment:
Why not use enum here instead of String constants?

Reply:
Yeah, I had thought about using enum, and that does seem to make more sense.  When I went looking for examples in the Java formatter being used in `formatter-maven-plugin`, it seemed to be using Strings for everything, so I went with that.  e.g. https://github.com/eclipse/eclipse.jdt.core/search?l=Java&q=brace_position_for_annotation_type_declaration

Happy to revisit, just let me know.

Reply:
I'm not really sure why they used Strings in their code everywhere... perhaps to make it easier to parse configuration files without converting keys in config files with `Enum.valueOf(...)`? Not sure. I'll leave it to you to decide which is best.

Reply:
I won't try to pretend I know which way is better here.  If you don't have an objection to leaving it as String, then I will just keep it this way.